### ISSUE TITLE
Tabs: add dismissable functionality to tab

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#12813]
This issue is for the implementation of dismissible functionality for tabs.

Open questions:
- There's lots of conversation around dismissible tabs with regards to accessibility:
  - https://github.com/w3c/aria/issues/1630
  - close button probably doesn't have to be in the keyboard nav, could be aria-hidden
  - but it could be if we'd like to give the user more affordance to discovery of the delete function
  - aria `tab` role spec says children are presentational only
- Does the design allow for users to put non-operable icon tabs next to dismissasble tabs in the tablist?
  - A: No, the entire tab group/list must be all the same type
- Will this actually unmount the tab component, or will it just be a handler?
  - We'll probably want to provide users a callback that they can use to unmount the component
  - This should support consumer's ability to add a confirmation dialog or similar before actually closing/unmounting the tab.
  - This also will need to support the use case of an tab containing unsaved work, to prompt for save, and/or allow devs to save data in the background
- Does this apply to ContainedTabs? Can they be dismissed?
  

https://github.com/carbon-design-system/carbon/issues/12813#tasklist-block-abb364de-c55b-4db3-93ea-8939819821a2

**Comment by francinelucca** [comment]
> Does the design allow for users to put non-operable icon tabs next to dismissasble tabs in the tablist?
> A: No, the entire tab group/list must be all the same type

@laurenmrice  @kingtraceyj  What's the expected behavior if this happens? what should be rendered if a user tries to mix dismissable tabs with non-operable tabs?

**Comment by francinelucca** [comment]
> [ ] Dismissible tabs: add delete/close via keyboard and operable 'x' icons

is the keyboard close interaction performed through focus > space/enter or just the Escape key?
CC @tay1orjones @mbgower

**Comment by tay1orjones** [comment]
@francinelucca The [WAI-ARIA Tabs pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) states <kbd>Delete</kbd> should be used, not <kbd>Escape</kbd>. 

All children elements of `tab` are presentational in nature and I believe should not receive focus, so this would additionally rule out the focus > space/enter.

**Comment by tay1orjones** [comment]
> What's the expected behavior if this happens? what should be rendered if a user tries to mix dismissable tabs with non-operable tabs?

I'd vote for dismissable to "win" and the icons on all other tabs are not rendered. With how composable the API is though we won't always be able to restrict this I think. It might be something we could warn about in dev mode though through conditional props or something?

**Comment by tay1orjones** [comment]
@francinelucca for clarity though, the spec does allow for tabs to be dismissable and have a descriptive icon https://github.com/carbon-design-system/carbon/issues/8487#issuecomment-1020664248, it's just that all tabs must be dismissable or not I think
